src/os/bluestore/BlueStore.cc

```diff
@@ -5266,30 +5266,13 @@ int BlueStore::_setup_block_symlink_or_file(
 	}
 
 	if (cct->_conf->bluestore_block_preallocate_file) {
-#ifdef HAVE_POSIX_FALLOCATE
-	  r = ::posix_fallocate(fd, 0, size);
-	  if (r) {
+          r = ::ceph_posix_fallocate(fd, 0, size);
+          if (r) {
```

Comment:
before your change `r` is the return code of `posix_fallocate()`, after your change, `r` is `-(-errono)`. is this what you expect?

Reply:
@tchaikov 
Good catch, if fumbled that one.

Reply:
@wjwithagen this is not addressed.